pkg/project/auth/cache.go

```diff
@@ -113,37 +122,27 @@ type AuthorizationCache struct {
 
 	reviewer Reviewer
 
-	namespaceInterface            kclient.NamespaceInterface
-	policyBindingsNamespacer      client.PolicyBindingsNamespacer
-	policiesNamespacer            client.PoliciesNamespacer
-	clusterPolicyBindingInterface client.ClusterPolicyBindingsInterface
-	clusterPolicyInterface        client.ClusterPoliciesInterface
-
 	syncHandler func(request *reviewRequest, userSubjectRecordStore cache.Store, groupSubjectRecordStore cache.Store, reviewRecordStore cache.Store) error
 }
 
 // NewAuthorizationCache creates a new AuthorizationCache
-func NewAuthorizationCache(reviewer Reviewer,
-	namespaceInterface kclient.NamespaceInterface, policyBindingsNamespacer client.PolicyBindingsNamespacer, policiesNamespacer client.PoliciesNamespacer,
-	clusterPolicyBindingInterface client.ClusterPolicyBindingsInterface, clusterPolicyInterface client.ClusterPoliciesInterface) *AuthorizationCache {
+func NewAuthorizationCache(reviewer Reviewer, namespaceInterface kclient.NamespaceInterface, policyClient policyclient.ReadOnlyPolicyClient) *AuthorizationCache {
 	result := &AuthorizationCache{
-		namespaceStore:                cache.NewStore(cache.MetaNamespaceKeyFunc),
-		policyIndexer:                 cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{"namespace": cache.MetaNamespaceIndexFunc}),
-		policyBindingIndexer:          cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{"namespace": cache.MetaNamespaceIndexFunc}),
-		clusterPolicyStore:            cache.NewStore(cache.MetaNamespaceKeyFunc),
-		clusterPolicyBindingStore:     cache.NewStore(cache.MetaNamespaceKeyFunc),
-		reviewRecordStore:             cache.NewStore(reviewRecordKeyFn),
-		userSubjectRecordStore:        cache.NewStore(subjectRecordKeyFn),
-		groupSubjectRecordStore:       cache.NewStore(subjectRecordKeyFn),
-		clusterBindingResourceVersion: "",
+		namespaceStore:                 cache.NewStore(cache.MetaNamespaceKeyFunc),
+		namespaceInterface:             namespaceInterface,
+		namespaceSyncResourceVersioner: &unchangingLastSyncResourceVersioner{},
+
 		clusterPolicyResourceVersion:  "",
-		namespaceInterface:            namespaceInterface,
-		policyBindingsNamespacer:      policyBindingsNamespacer,
-		policiesNamespacer:            policiesNamespacer,
-		clusterPolicyBindingInterface: clusterPolicyBindingInterface,
-		clusterPolicyInterface:        clusterPolicyInterface,
-		reviewer:                      reviewer,
-		skip:                          &neverSkipSynchronizer{},
+		clusterBindingResourceVersion: "",
```

Comment:
can you make clusterBindingResourceVersion and clusterPolicyResourceVersion a util.StringSet?

would like to just see this be multi-object aware, so we can remove the TODO.